cohere_learn/classify.py

```diff
@@ -0,0 +1,190 @@
+"""
+Classification Utils built over Cohere API
+
+Raises:
+    ValueError: In cases of invalid test counts or other size mismatches
+"""
+from pathlib import Path
+from typing import List, Tuple
+
+import cohere
+import pandas as pd
+from cohere.classify import Example
+from tqdm.notebook import tqdm
+
+Path.ls = lambda x: list(x.iterdir())  # type: ignore
+
+
+class CohereBase:
+    """
+    Base Utils for the Cohere API
+    """
+
+    def make_examples(self, df: pd.DataFrame) -> List[Example]:
+        """
+        #TODO: Iterative and slow, make this parallel and fast
+        """
+        examples = []
+        for row in df.iterrows():
+            text = row[1][self.x_label]
+            lbl = row[1][self.y_label]
+            examples.append(Example(text, lbl))
+        return examples
+
+    @staticmethod
+    def parse_classification(
+        classification: cohere.classify.Classification,
+    ) -> Tuple[str, float]:
+        """
+        Parse a single classification from Cohere Response into a tuple of (label, score)
+
+        Args:
+            classification (cohere.classify.Classification): Single classification object
+
+        Returns:
+            Tuple[str, float]: lbl, score for a single classification
+        """
+        lbl = classification.prediction
+        confidences = classification.confidence
+        score = -1
+        for confidence in confidences:
+            score = max(score, confidence.confidence)
+            # why is this max? lazy hack since there is no prediction confidence score in the response
```

Comment:
this could be done with 
`response.classifications[0].labels[response.classifications[0].prediction].confidence` 
and when we push the new version of the sdk it will be able to be done just with 
`response.classifications[0].confidence`

Reply:
Aah, yes. That's cleaner. Looking forward to the new version!

Reply:
Made the changes to make the entire SDK `response` parsing in one place and moving `Example` usage inside specific implementations